File: R/forLater/David_detectors/rates_test.R
rates_test <- function(data, sampling_rate, FUN, bktime, indices, events, ntests, testint) {
  tpevents <- ((indices/sampling_rate)/bktime)
  sr <- sampling_rate
  for (k in 1:ntests) {
    if (k == 1) {
      thresh <- testint
    } else {
      if (k == ntests) {
        thresh <- max(njerk(data, sampling_rate))
      }
    }
    detections <- detect(data, sr, FUN, thresh, bktime, plot_peaks = FALSE, sampling_rate=sampling_rate)$peak_time
    True_Positive_Rate <- acc_test(detections, events, sampling_rate, tpevents)$hits_rate
    False_Positive_Rate <- acc_test(detections, events, sampling_rate, tpevents)$false_alarm_rate
    thresh <- thresh + testint
    if (k == 1) {
      pts <- rbind(c(0,0), c(True_Positive_Rate,False_Positive_Rate))
    } else {
      pts <- rbind(pts, c(True_Positive_Rate,False_Positive_Rate))
    }
  }
  pts <- rbind(pts, c(1,1))
  True_Positive_Rate <- pts[, 1]
  False_Positive_Rate <- pts[, 2]
  rates <- cbind(True_Positive_Rate, False_Positive_Rate)
  return(rates)
}

File: matlab/not_for_distribution/David_detectors/rates_test.m
function pts = rates_test(data, sampling_rate, FUN, bktime, indices, events, ntests, testint) 
  
tpevents = ((indices/sampling_rate)/bktime);
sr = sampling_rate;

for k = 1:ntests
    if k == 1
        thresh = testint;
    elseif k == ntests
        thresh = max(njerk(data, sampling_rate));
    end
    detections = detect(data, sr, FUN, thresh, bktime, false, sampling_rate);
    detections = detections.peak_time;
    True_Positive_Rate = acc_test(detections, events, sampling_rate, tpevents);
    True_Positive_Rate = True_Positive_Rate.hits_rate;
    False_Positive_Rate = acc_test(detections, events, sampling_rate, tpevents);
    False_Positive_Rate = False_Positive_Rate.false_alarms_rate;
    thresh = thresh + testint;
    if k == 1
        pts = [0,0; True_Positive_Rate, False_Positive_Rate];
    else
        pts = [pts; True_Positive_Rate, False_Positive_Rate];
    end
end

pts = [pts; 1,1];